### ISSUE TITLE
Input command priority not respected after config reload

### ISSUE THREAD

**Issue by 9ary** [swaywm/sway#7271]
- **Sway Version:** 1.7

- **Debug Log:** [sway.log](https://wank.party/JuPQ.log)

- **Configuration File:**
```
input "1390:269:ELECOM_TrackBall_Mouse_HUGE_TrackBall" {
accel_profile adaptive
middle_emulation enabled
pointer_accel -0.25
scroll_button 279
scroll_method on_button_down
}

input "type:pointer" {
accel_profile flat
}
```

- **Description:**

After starting sway, acceleration works as expected on the trackball. After a configuration reload, accel_profile is set back to flat until I change it via swaymsg.

Reordering the configuration does solve the problem, but I can't really control that because I generate it with home-manager, and the manual says this should work anyway.

> In the configuration file, settings with a more specific selector take precedence over more general ones: _\<identifier\>_ \> _type:\<input_type\>_ \> _\*_.  When executing input commands, however, the settings are applied to all matching input devices!  This means that _type:\<input_type\>_ can override previously set _\<identifier\>_ settings, even though in a configuration file they would take precedence.  Similarly _\*_ can override both _\<identifier\>_ and _type:\<input_type\>_ settings, if applied later.

**Comment by bl4ckb0ne** [comment]
Have you tried swapping them? Set the for all pointers then for your device.

**Comment by 9ary** [comment]
Yes, swapping the two sections does fix it, I said so in the op, but maybe I was unclear. Unless you mean something else?

**Comment by bl4ckb0ne** [comment]
Ah i missed it sorry. It shouldn't work because the config file is read top to bottom, setting a specific device will always be overriden by a generic config.

**Comment by 9ary** [comment]
That's in direct conflict with what the manual says, and it works perfectly fine from a cold start. It only breaks when reloading.

The config file being order-sensitive makes perfect sense to me, but in that case the manual needs to reflect that _and_ the code should be fixed to have consistent behavior between cold starts and reloads.

**Comment by pacien** [comment]
I'm experiencing the same issue.

```
input "7764:8240:TypeMatrix.com_USB_Keyboard" {
xkb_layout fr
xkb_model pc105
xkb_numlock enabled
xkb_variant bepo
}

input "type:keyboard" {
xkb_layout fr
xkb_numlock enabled
xkb_variant oss
}
```

On a fresh start, the rules are applied with the priority given to the most
precise one as documented in the manual page.

On reload, the priority follows the order of appearance in the file,
contradicting the manual page.

Like the author of this issue, I cannot control this order because my
configuration file is generated, so this is problematic.
